pycqed/analysis_v2/spectroscopy_analysis.py

```diff
@@ -31,1507 +23,285 @@ def __init__(self, t_start: str = None,
-                                    'k--', linewidth=1.5)
-
-                            par = ["%.3f" %(fit_results[0].params['gamma_ro'].value*1e-6),
-                                   "%.3f" %(fit_results[0].params['omega_pf'].value*1e-9),
-                                   "%.3f" %(fit_results[0].params['kappa_pf'].value*1e-6),
-                                   "%.3f" %(fit_results[0].params['J'].value*1e-6),
-                                   "%.3f" %(fit_results[0].params['omega_ro'].value*1e-9),
-                                   "%.3f" %(fit_results[1].params['omega_ro'].value*1e-9),
-                                   "%.3f" %((fit_results[1].params['omega_ro'].value-
-                                             fit_results[0].params['omega_ro'].value)
-                                             /2*1e-6)]
-                            textstr = str('\n\nkappa = '+par[2]+' MHz'
-                                          +'\n\nJ = '+par[3]+' MHz'
-                                          +'\n\nchi = '+par[6]+' MHz'
-                                          +'\n\nf_pf = '+par[1]+' GHz'
-                                          +'\n\nf_rr |g> = '+par[4]+' GHz'
-                                          +'\n\nf_rr |e> = '+par[5]+' GHz'
-                                          +'\n\nf_RO = '+"%.3f" %(f_RO*1e-9)+''
-                                          ' GHz'
-                                         )
-                            ax.plot([f_RO],
-                                    [0],
-                                    'w--', label=textstr)
-                        # box_props = dict(boxstyle='Square',
-                        #                  facecolor='white', alpha=0.8)
-                        # self.box_props = {key: val for key,
-                        #                                val in box_props.items()}
-                        # self.box_props.update({'linewidth': 0})
-                        # self.box_props['alpha'] = 0.
-                        #
-                        ax.legend(loc='upper left', bbox_to_anchor=[1, 1])
-
-                else:
-                    reso_freqs = [fit_results[tt].params['f0'].value *
-                                  1e9 for tt in range(len(self.raw_data_dict['timestamps']))]
-                    ax.plot(np.squeeze(self.plot_xvals),
-                            reso_freqs,
-                            'o',
-                            color='m',
-                            markersize=3)
-
-    def plot(self, key_list=None, axs_dict=None, presentation_mode=None, no_label=False):
-        super(ResonatorSpectroscopy, self).plot(key_list=key_list,
-                                                axs_dict=axs_dict,
-                                                presentation_mode=presentation_mode)
-        if self.do_fitting:
-            self.plot_fitting()
-
-
-class ResonatorSpectroscopy_v2(Spectroscopy):
```

Comment:
You cannot remove this, it is being used in QuDev_transmon.find_readout_frequency(). This is the routine we use to find the optimal readout frequency.

Reply:
Also check that these classes still work after your changes to BaseDataAnalysis and other changes in spectroscopy_analysis.py

Reply:
Finally, did you check that the changes in BaseDataAnalysis do not break MultiQubit_TimeDomain_Analysis?